controller/src/main/java/org/jboss/as/controller/ListAttributeDefinition.java

```diff
@@ -61,6 +63,10 @@ public ParameterValidator getElementValidator() {
         return elementValidator;
     }
 
+    public AttributeDefinition getValueType() {
```

Comment:
Could you add some javadoc please?

Why would the valuetype of list throws an exception about YAML configuration?

Reply:
Ok

Reply:
my questions have not been answered afaict?

Reply:
It doesn't throw a YAML configuration exception anymore

Reply:
Please keep in mind that this class is in the public API and can be used by any extension.

I'd not expect to get an unsupported exception from this base class. 
If only specialised subclass can support the YAML feature, let's move the corresponding API to their levels

Reply:
Our usual convention to put "experimental" stuff in our public API is to `@Deprecate` the method with a warning that this may be removed in future releases e.g. https://github.com/wildfly/wildfly-core/blob/0d189892dd2a503baed3a38989d2b2c21d0ab5c6/server/src/main/java/org/jboss/as/server/ExternalManagementRequestExecutor.java#L47-L51

Reply:
This class is abstract, the gaol was to provide a 'default' behavior

Reply:
sigh... so we have:

* ListAttributeDefinition that now defines `AttributeDefinition getValueType()`
* PrimitiveListAttributeDefinition.java  that used to define `ModelType getValueType()`
* SimpleListAttributeDefinition.java  that used to define `AttributeDefinition getValueType()`

is that correct?

There are issues with these commits:

* PrimitiveListAttributeDefinition#getValueType is a public method. You can't change its return type like that.
* The default behaviour for a list attribute definition is to throw an exception that there is no value type. This is fishy to me...

Reply:
for the1st point I agree (you could have added ObjectListAttributeDefinition to the list)
for the 2nd I disagree, throwing an UnsupportedOperationException when a method is not implemented is not something fishy. We have it at several places in our codebase for default methods in interfaces for example